snooty/postprocess.py

```diff
@@ -648,7 +702,45 @@ def get_include_argument(node: n.Directive) -> str:
             assert include_page is not None
             ast = include_page.ast
             assert isinstance(ast, n.Parent)
-            node.children = [util.fast_deep_copy(ast)]
+
+            deep_copy_children: MutableSequence[n.Node] = [util.fast_deep_copy(ast)]
+
+            # TODO: Move subsgraphing implementation into parse layer, where we can
+            # ideally take subgraph of the raw RST
+            start_after_text = (
+                node.options["start-after"] if "start-after" in node.options else ""
+            )
+            end_before_text = (
+                node.options["end-before"] if "end-before" in node.options else ""
+            )
+
+            if start_after_text or end_before_text:
+                try:
+                    # Returns a subgraph of the AST based on text bounds
+                    deep_copy_children, any_start, any_end = self.bound_included_AST(
+                        deep_copy_children, start_after_text, end_before_text
+                    )
+                    # Confirm that we found all specified text (with helpful diagnostic message if not)
+                    msg = (
+                        "If you are sure your text is within the included RST file, try using it in a comment or "
+                        "label. Search is case-sensitive."
```

Comment:
[nit] Newline ➡️ good use case for an [f-string](https://www.python.org/dev/peps/pep-0498/).

Reply:
Note: leaving as regular string due to lint warning: `"f-string is missing placeholders"`